src/Collection.php

```diff
@@ -94,7 +96,13 @@ public static function range($start = 0, $end = null, $step = 1)
     public function getIterator()
     {
         if ($this->inputFactory) {
-            $this->input = call_user_func($this->inputFactory);
+            $input = call_user_func($this->inputFactory);
+
+            if (is_array($input)) {
+                $input = new RecursiveArrayIterator($input);
+            }
+
```

Comment:
@DusanKasan Having said that, I didn't get any good idea in spite of I was thinking for a minute. Sorry, never mind and just ignore.